furusiya/model/components/ai/monster.py

```diff
@@ -39,17 +50,17 @@ def take_turn(self):
         else:
             # restore the previous AI (this one will be deleted because it's not
             # referenced anymore)
-            self.owner.ai = self.owner.original_ai
+            self.owner.set_component(self.original_ai)
             message('The ' + self.owner.name + ' is no longer stunned!', colors.red)
             self.owner.char = self.owner.name[0]
 
 
-class ConfusedMonster:
+class ConfusedMonster(TemporaryAI):
```

Comment:
Just reading this class statement makes me into a `ConfusedReviewer`. Why is this extending `TemporaryAI`?

Monster AIs should never know about statuses. That's a separate concern.